src/index.js

```diff
@@ -221,15 +196,12 @@ MyArray.prototype.slice = function(begin, end) {
 };
 
 MyArray.prototype.find = function(callback, thisArg) {
-  let element = null;
-
   for (let i = 0; i < this.length; i++) {
-    element = this[i];
-
-    if (callback.call(thisArg, element, i, this)) {
-      return element;
+    if (callback.call(thisArg, this[i], i, this)) {
+      return this[i];
     }
   }
+  return undefined;
```

Comment:
You may not return undefined. If there is no match found, undefined would be returned in any case.